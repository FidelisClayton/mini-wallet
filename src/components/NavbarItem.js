import React from 'react'
import { Link } from 'react-router-dom'

const NavbarItem = ({
  to,
  text,
  active
}) => {
  const classNames = [
    'nav__link',
    active && 'nav__link--active'
  ]
    .filter(Boolean)
    .join(' ')

  return (
    <li className="nav__item">
      <Link
        to={to}
        className={classNames}
      >
        { text }
      </Link>
    </li>
  )
}

export default NavbarItem
