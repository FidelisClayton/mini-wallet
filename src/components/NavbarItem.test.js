import React from 'react'
import { shallow } from 'enzyme'

import NavbarItem from './NavbarItem'

describe('Component: <NavbarItem />', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <NavbarItem
        to={'/home'}
        text={'Home'}
        active={true}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
