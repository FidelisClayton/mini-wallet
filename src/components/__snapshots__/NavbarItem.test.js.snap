// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <NavbarItem /> should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavbarItem
    active={true}
    text="Home"
    to="/home"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="nav__link nav__link--active"
        replace={false}
        to="/home"
>
        Home
</Link>,
      "className": "nav__item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Home",
        "className": "nav__link nav__link--active",
        "replace": false,
        "to": "/home",
      },
      "ref": null,
      "rendered": "Home",
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="nav__link nav__link--active"
          replace={false}
          to="/home"
>
          Home
</Link>,
        "className": "nav__item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Home",
          "className": "nav__link nav__link--active",
          "replace": false,
          "to": "/home",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
